src/controller/discoverymgr/mnedc/tunmgr/tun.go

```diff
@@ -53,15 +54,29 @@ func GetInstance() Tun {
 
 //CreateTUN creates a virtual tun interface
 func (TunImpl) CreateTUN() (*water.Interface, error) {
-	intf, err := water.NewTUN("")
+	intf, err := water.New(water.Config{
+		DeviceType: water.TUN,
+	})
+
 	return intf, err
 }
 
 //SetTUNIP sets an IP to the interface
 func (TunImpl) SetTUNIP(iName string, localAddr net.IP, addr *net.IPNet, debug bool) error {
-	sargs := fmt.Sprintf("%s %s netmask %s", iName, localAddr.String(), net.IP(addr.Mask).String())
-	args := strings.Split(sargs, " ")
-	return commandExec("ifconfig", args, debug)
```

Comment:
Thank you for changing these lines of code. This is a potential security vulnerability.